pkg/testutils/lint/lint_test.go

```diff
@@ -2794,4 +2796,51 @@ func TestLint(t *testing.T) {
 			}
 		}
 	})
+
+	// Test forbidden roachtest imports.
+	t.Run("TestRoachtestForbiddenImports", func(t *testing.T) {
+		t.Parallel()
+
+		roachprodLoggerPkg := "github.com/cockroachdb/cockroach/pkg/roachprod/logger"
+		// forbiddenImportPkg -> permittedReplacementPkg
+		forbiddenImports := map[string]string{
+			"github.com/cockroachdb/cockroach/pkg/util/log": roachprodLoggerPkg,
+			"log": roachprodLoggerPkg,
+		}
+
+		// grepBuf creates a grep string that matches any forbidden import pkgs.
+		var grepBuf bytes.Buffer
+		grepBuf.WriteByte('(')
+		for forbiddenPkg := range forbiddenImports {
+			grepBuf.WriteByte('|')
+			grepBuf.WriteString(regexp.QuoteMeta(forbiddenPkg))
+		}
+		grepBuf.WriteString(")$")
+
+		filter := stream.FilterFunc(func(arg stream.Arg) error {
+			for _, pkg := range pkgs {
+				for _, s := range pkg.Imports {
+					arg.Out <- pkg.PkgPath + ": " + s.PkgPath
+				}
+			}
+			return nil
+		})
+		if err := stream.ForEach(stream.Sequence(
+			filter,
+			stream.Sort(),
+			stream.Uniq(),
+			stream.Grep(`cockroach/pkg/cmd/roachtest/tests`),
```

Comment:
One gotcha with linters of this type is they can fail silently, e.g., one day we decide to refactor roachtests and move them to a different package. One small addition could be a check to see that we actually analyzed non-zero sources, or otherwise, raise an error.

Reply:
E.g.,

```
numAnalyzed := 0
		if err := stream.ForEach(stream.Sequence(
			filter,
			stream.Sort(),
			stream.Uniq(),
			stream.Grep(`cockroach/pkg/cmd/roachtest/tests:`),
		), func(s string) {
			pkgStr := strings.Split(s, ": ")
			_, importedPkg := pkgStr[0], pkgStr[1]
			numAnalyzed++

			// Test that a disallowed package is not imported.
			if replPkg, ok := forbiddenImports[importedPkg]; ok {
				t.Errorf("\n%s <- please use %q instead of %q", s, replPkg, importedPkg)
			}
		}); err != nil {
			t.Error(err)
		}
		if numAnalyzed == 0 {
			t.Errorf("Empty input! Please check the linter.")
		}
```

Reply:
Great idea! I'll add both suggestions from above.

Reply:
Done.